build/azDevOps/azure/ci-cosmosdb-worker-vars.yml

```diff
@@ -0,0 +1,172 @@
+
+  # Stacks operates Terraform states based on workspaces **IT IS VERY IMPORTANT** that you ensure a unique name for each application definition
+  # Furthermore **IT IS VERY IMPORTANT** that you change the name of a workspace for each deployment stage
+  # there are some best practices around this if you are going for feature based environments
+  # - we suggest you create a runtime variable that is dynamically set based on a branch currently running
+  # **`terraform_state_workspace: `**
+  # avoid running anything past dev that is not on master
+  # sample value: company-webapp
+  - name: tf_state_rg
+    value: {{ .Input.Terraform.Backend.Group }}
+  - name: tf_state_storage
+    value: {{ .Input.Terraform.Backend.Storage }}
+  - name: tf_state_container
+    value: {{ .Input.Terraform.Backend.Container }}
+  - name: tf_state_key
+    value: $(self_generic_name)
+  - name: TF_FILE_LOCATION
+    value: deploy/azure/app/kube
+
+  # App Specific Taxonomy
+  - name: resource_def_name
+    value: $(self_generic_name)
+  - name: component
+    value: cosmosdb-worker
+  - name: role
+    value: backend
+  - name: k8s_app_route
+    value: /api/menu
+  - name: app_name
+    value: $(self_generic_name)
+
+  # Agent configuration
+  - name: TaskctlVersion
+    value: 1.5.2
+  - name: pool_vm_image
+    value: ubuntu-20.04
+
+  # SelfConfig
+  # If you haven't specified source_repo at cli runtime please ensure you replace it here
+  # It is case sensitive for TFS based repos
+  - name: self_repo_src
+    value: src/func-cosmosdb-worker/src/functions
+  - name: self_repo_dir
+    value: "$(Agent.BuildDirectory)/s/$(self_repo)"
+  - name: self_repo_tf_src
+    value: deploy/azure/app/kube
+  - name: self_repo_tf_dir
+    value: "$(self_repo_dir)/$(self_repo_tf_src)"
+
+  # SonarQube Configuration
+  # SONAR_TOKEN must be inherited from library groups/etc.
+  - name: sonar_organization
+    value: {{ .Input.Business.Company }}
+  - name: sonar_url
+    value: "https://sonarcloud.io"
+  - name: sonar_properties
+    value: '/d:sonar.cs.opencover.reportsPaths="/app/coverage/*.opencover.xml" /d:sonar.exclusions="**/*Tests" /d:sonar.coverage.exclusions="**/*Tests"'
+  - name: sonar_project_name
+    value: {{ .Project.Name }}
+
+  # Test Configuration
+  # - name: FUNCTIONAL_TESTS_SRC_DIR
+  #   value: src/func-cosmosdb-worker/src/tests/Functional
+  # - name: FUNCTIONAL_TESTS_RUN_DIR
+  #   value: /app/tests
+  # - name: FUNCTIONAL_TEST_BASEURL
+  #   value: https://$(Environment.ShortName)-$(domain).$(dns_base_domain)$(k8s_app_route)/
+
+  # Docker Container registry configuration (used for image tagging and pushing)
+  # TODO: Despite being available in the TF State, these are required in other steps
+  # TODO: which do not inherit the state variables yet
+  # TODO: Promotion is now doing nothing as same registry for both envs, consider removing or creating another registry for Prod
+  - name: docker_container_registry_rg_name_nonprod
+    value:
```

Comment:
Do these need values?

Reply:
No, these need to be populated by the end user. I do think a lot of these need some more thought, we generate most of the resources but don't currently have a way of outputting the details for later steps